jax/_src/lax/control_flow.py

```diff
@@ -3122,4 +3130,39 @@ def _remat_translation_rule(*args,
     mlir.register_lowering(remat_primitive,
                            mlir.lower_fun(partial(_remat_translation_rule,
                                                    platform=platform),
-                                          multiple_results=True))
+                                          multiple_results=True),
+                           platform=platform)
+
+
+def _optimization_barrier_abstract_eval(*args):
+  return args
+
+
+def _optimization_barrier_translation_rule(ctx, avals_in, avals_out, *args):
+  out = xops.OptimizationBarrier(xops.Tuple(ctx.builder, args))
+  return [xops.GetTupleElement(out, i) for i in range(len(args))]
+
+
+def _optimization_barrier_lowering_rule(ctx, *args):
+  loop_carry_types = _map(mlir.aval_to_ir_types, ctx.avals_in)
```

Comment:
I suspect `loop_carry` is a poor choice of name here, given there are no longer any loops.

Reply:
Done.